include/envoy/http/codec.h

```diff
@@ -132,6 +142,14 @@ class Stream {
    * @param reason supplies the reset reason.
    */
   virtual void resetStream(StreamResetReason reason) PURE;
+
+  /**
+   * Enable/disable further data from this stream.
+   * Cessation of data may not be immediate.  For example, for HTTP/2 this may stop further flow
+   * control window updates which will result in the peer eventually stopping sending data.
```

Comment:
For a later implementation, have you thought about using distance from watermark as a way to size the H2 stream flow control window?